src/features/content-explorer/content-explorer/ContentExplorer.js

```diff
@@ -329,9 +331,10 @@ class ContentExplorer extends Component {
     };
 
     handleItemClick = ({ event, index }) => {
-        const { contentExplorerMode, items, onSelectItem, onSelectedItemsUpdate } = this.props;
+        const { contentExplorerMode, items, onSelectItem, onSelectedItemsUpdate, controlledSelectedItems } = this.props;
         const { selectedItems } = this.state;
         const item = items[index];
+        const allSelectedItems = { ...selectedItems, ...controlledSelectedItems };
```

Comment:
I was just wondering, since you're doing this in few place, maybe instead create a method:
```
    getAllSelectedItems = () => {
        return { ...this.state.selectedItems, ...this.props.controlledSelectedItems };
    };
 ```